helios-client/src/main/java/com/spotify/helios/client/DefaultRequestDispatcher.java

```diff
@@ -77,20 +73,16 @@
 
   private static final Logger log = LoggerFactory.getLogger(DefaultRequestDispatcher.class);
 
-  private static final long RETRY_TIMEOUT_MILLIS = SECONDS.toMillis(60);
   private static final long HTTP_TIMEOUT_MILLIS = SECONDS.toMillis(10);
-  private static final List<String> VALID_PROTOCOLS = ImmutableList.of("http", "https");
-  private static final String VALID_PROTOCOLS_STR =
-      String.format("[%s]", Joiner.on("|").join(VALID_PROTOCOLS));
 
-  private final Supplier<List<URI>> endpointSupplier;
+  private final Iterator<Endpoint> endpointIterator;
```

Comment:
You can only iterate over the items in an Iterator once. Does this mean a new instance of DefaultRequestDispatcher has to be made for each request?